modules/xmpp/ChatRoom.js

```diff
@@ -292,7 +307,21 @@ export default class ChatRoom extends Listenable {
             if (meetingIdValEl.length) {
                 this.setMeetingId(meetingIdValEl.text());
             } else {
-                logger.trace('No meeting ID from backend');
+                logger.warn('No meeting ID from backend');
+            }
+
+            const membersOnly = $(result).find('>query>feature[var="muc_membersonly"]').length === 1;
+
+            if (membersOnly !== this.membersOnlyEnabled) {
+                this.membersOnlyEnabled = membersOnly;
+                this.eventEmitter.emit(XMPPEvents.MUC_MEMBERS_ONLY_CHANGED, membersOnly);
+            }
+
+            const lobbyRoomField
+                = $(result).find('>query>x[type="result"]>field[var="muc#roominfo_lobbyroom"]>value');
+
+            if (lobbyRoomField.length && this.lobby) {
+                this.lobby.setLobbyRoomJid(lobbyRoomField.text());
```

Comment:
Actually it doesn't matter ... 
When we do setLobbyRoomJid is when we join the lobby room. 
MUC_MEMBERS_ONLY_CHANGED is when someone set it to true or false ... remotely ... and we sync the state ... I see it make sense mainly when disabling ... I will do some further testing to make sure we do not send the form once again in case of multiple moderators.
In general order of these two does not matter, at least in current code.

Reply:
I moved it around as you suggested and changed a method name, it was anyway doing form submit which was not needed.